extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/SparkSessionProvider.scala

```diff
@@ -71,23 +75,78 @@ trait SparkSessionProvider {
 
   protected val sql: String => DataFrame = spark.sql
 
-  protected def doAs[T](user: String, f: => T): T = {
+  protected def doAs[T](user: String)(f: => T): T = {
     UserGroupInformation.createRemoteUser(user).doAs[T](
       new PrivilegedExceptionAction[T] {
         override def run(): T = f
       })
   }
+
+  protected def runSqlAs(user: String)(
```

Comment:
I think it's kind of over-engineering, which increases the onboarding costs for new developer

Reply:
+1

Reply:
It's not forcing user/developer to use `runSqlAs`. `runSqlAs` and `runSqlAsWithAccessException` are together providing a concentrative and declarative way in most general scenarios to decide running what SQL in which user with asserting what messages. `doAs` is still good to be used with them,  and remained as you see.

Reply:
I have added a `checkAnswer` method and it looks better than this one. Meanwhile, there would be great if we could have a similar one for expected failures, for example.

```scala
def checkError[T <: Throwable : ClassTag](
  user: String,
  query: String,
  expectedErrMsg: String): Unit = {
...
  }
```

Reply:
It's not sure `checkAnswer` is better in some way.

As for this PR, the key points for changes in signatures of `doAs` and `runAs` includes:
- extract user as first param group for `As`, to focus in the user idendity and also to eliminate unhelpfule line break every time
- handy to config the decision points in assertion and and the execution modes
- wrapping repeated exectution template

After all, these methods are used in test cases only. It's good enough to provide well polished signatures for simplication.

Reply:
Please use simple functions for tests, especially in test traits/helpers, etc.

- Complex helpers make the debug experience worse.
- Playing tricks with if-else, etc., make the function itself hard to understand. while you wrote once, thousands of reads may follow.
- Currying is not so that welcome in tests
- etc